frontend_tests/node_tests/integration_bot_widget.js

```diff
@@ -0,0 +1,50 @@
+set_global('page_params', {
+    is_admin: false,
+    realm_users: [],
+});
+
+set_global('$', function () {
+});
+
+add_dependencies({
+    people: 'js/people.js',
+    stream_data: 'js/stream_data.js',
+});
+
+set_global('blueslip', {});
+
+var integration_bot_widget = require('js/integration_bot_widget.js');
+
+(function test_set_integration_bot_url() {
+    var external_api_uri_subdomain = "https://chat.zulip.org/api/v1/external/";
+    var integration_url = "airbrake";
+    var bot_full_name = "airbrake";
+    var bot_short_name = "airbrake";
+    var bot_avatar_file; // I am not sure how to point this variable to a file.
+    var stream_name = "Airbrake";
+
+    var airbrake = {
+        subscribed: false,
+        color: 'blue',
+        name: 'Airbrake',
+        stream_id: 1,
+        in_home_view: false,
+    };
+
+    // Add stream to sub
+    stream_data.add_sub(stream_name, airbrake);
+
+    var bot = {
+        bot_full_name: bot_full_name,
+        bot_short_name: bot_short_name,
+        bot_avatar_file: bot_avatar_file,
+        stream_name: stream_name,
+        external_api_uri_subdomain: external_api_uri_subdomain,
+        integration_url: integration_url,
+    };
+
+    integration_bot_widget.set_integration_bot_url(bot);
+
+    var integration_bot_url; // ?
```

Comment:
To check if the URL returned is correct or not, we need to have the bot api key to assert our result. But the bot api would be created in the call to create_bot.

How do I assert this call? Should I create a mock function that returns a specific api key (for testing purpose) ?